osu.Game/IO/FileStore.cs

```diff
@@ -70,43 +70,49 @@ public FileInfo Add(Stream data, bool reference = true)
                 }
 
                 if (reference || existing == null)
-                    Reference(info);
+                    Reference(ref info);
 
                 return info;
             }
         }
 
         public void Reference(params FileInfo[] files)
         {
-            if (files.Length == 0) return;
+            for (var i = 0; i < files.Length; i++)
+            {
+                Reference(ref files[i]);
+            }
+        }
 
+        public void Reference(ref FileInfo file)
```

Comment:
It won't going to work. Since original `FileInfo` has been tracked, but modifications to the found one won't affect the original one, which will lead to multiple `FileInfo` with same ID coexist in one entity and cause tracking conflicts.

Reply:
I also had a similar implementation to this in my attempt to fix this. I don't think it's too bad.